### DISCUSSION TITLE
Has anyone had success with onnx convert on espnet models?

### DISCUSSION THREAD

**Discussion by miziworld** [espnet/espnet#4230] [Q&A]
Several developers are requesting onnx convert to speed up espnet. Problems posted on Issuse are usually about input_dummy. Is there a developer with a solution for this? ( In my case, for ASR system )

Also, in order to put a random dummy_input, the format of the input has to be adjusted when the model is executed (eg, the arguments entered in *x in self.model(*x)), how can I check this?

**Comment by kan-bayashi** [comment]
This repository may help you.
https://github.com/Masao-Someki/espnet_onnx

**Reply by miziworld** [reply]
Thank you, I will try it.
However, to upload the model to huggingface, I ran steps 14 to 16 of asr.sh, the zip file is created in the exp path, but the model is not uploaded to the huggingface account.
I followed these steps and ran
```
./run.sh --stage 15 --skip_upload_hf false --hf_repo my/ksponspeech_VGGRNN
```
![image](https://user-images.githubusercontent.com/17849068/161422791-f2d47478-5038-4b25-9b5d-1ae4a5aefce4.png)
No models have been uploaded to my huggingface account. Do you know any solution for this?

**Reply by kan-bayashi** [reply]
Sorry I have never uses huggingface repo :(

**Comment by gowtham1997** [comment]
I think you could use the espnet_onnx repo without uploading the model to hf. 

```python
from espnet2.bin.asr_inference import Speech2Text
import os
import yaml
from espnet_onnx.export import ModelExport
from pathlib import Path

m = ModelExport()

os.chdir("egs2/librispeech_100/asr1/")


transducer_conf = yaml.safe_load(Path('conf/decode_rnnt_conformer.yaml').read_text())

speech2text = Speech2Text(asr_train_config="exp/asr_train_rnnt_conformer_raw_en_bpe5000_sp/config.yaml",
                          asr_model_file="exp/asr_train_rnnt_conformer_raw_en_bpe5000_sp/latest.pth",
                          transducer_conf=transducer_conf["transducer_conf"],
                          lm_weight=0.0)

m.export(speech2text, 'onnx_export', quantize=True)
```
^ I was running into some error in the last line as the transducer is not yet supported. But it might work in your case

**Reply by miziworld** [reply]
Thank you very much!
I will try it :-)
